Configuration/NativeMethods.cs

```diff
@@ -64,7 +64,9 @@ private struct AssemblyInfo
         public static string GetAssemblyPath(string name)
         {
             if (name == null)
-                throw new ArgumentNullException("name");
+                throw new ArgumentNullException("name");
+              
+            if( Type.GetType( "Mono.Runtime" ) != null ) return null;
 
```

Comment:
Also I don't think this type will be defined when running on ASP.NET Core CLR on Linux or Mac either, so skipping all logic in `AzureApplicationSettings` if you're not on windows seems best.